### DISCUSSION TITLE
Limiting populated entities for array of entities

### DISCUSSION THREAD

**Discussion by squivix** [mikro-orm/mikro-orm#5189] [Q&A]
I saw in pull request #1502  that you can use the `matching()` method of collections to limit the number of populated to-many entities for a single entity:

```ts
const a = await em.findOneOrFail(Author, 1);

// only loading the list of items
const books = await a.books.matching({ limit: 3, offset: 10, orderBy: { title: 'asc' } });
console.log(books); // [Book, Book, Book]
console.log(a.books.isInitialized()); // false

// storing the items in collection
const tags = await books[0].tags.matching({
  limit: 3,
  offset: 5,
  orderBy: { name: 'asc' },
  store: true,
});
console.log(tags); // [BookTag, BookTag, BookTag]
console.log(books[0].tags.isInitialized()); // true
console.log(books[0].tags.getItems()); // [BookTag, BookTag, BookTag]
```

But what about limiting populated entities for an array of entities? For example if you want to populate no more than 3 tags for ALL of the`books` in one query.

Maybe we could have the `limit` and `offset` options in `em.populate()` as discussed in #1059

**Comment by B4nan** [comment]
How do you expect that to work?

**Reply by squivix** [reply]
I have defintely not thought about it. I figured it was similar to how pagination is currently implemented.

Ok if it's too hard to implement then forget about it. I guess I'll do many queries with `matching()`.
